sphinx/ext/autosectionlabel.py

```diff
@@ -29,6 +29,17 @@ def get_node_depth(node: Node) -> int:
     return i
 
 
+def get_node_parents(node: Node) -> list[str]:
```

Comment:
```suggestion
def _find_parent_nodes(node: Node) -> list[str]:
```

Reply:
I followed the same naming convention as used in the get_node_depth function above.
Do you want me to change that one as well to follow the same convention, i.e. rename it to _get_node_depth ?

Reply:
I see where you're coming from. I'd prefer not bothering with keeping the naming convention. Historically, private and public APIs were named similarly, just that private ones were not documented. However, for new code, we're trying to avoid having public names if we can, especially for helper functions, so either `_get_parent_nodes` or `_find_parent_nodes` would be correct. 

- `_get_parent_nodes` would return a list
- `_find_parent_nodes` would yield elements one by one

Instead of constructing the list in the function itself, you can yield nodes one by one, hence the naming suggestion.